test/src/upgrade_eks_test.go

```diff
@@ -0,0 +1,230 @@
+// and check that everything is working as expected
+func (suite *UpgradeEKSTestSuite) TestUpgradeEKS() {
+	// create the eks cluster
+	suite.varTf = map[string]interface{}{
+		"name":                  suite.clusterName,
+		"region":                suite.region,
+		"np_desired_node_count": suite.expectedNodes,
+		"kubernetes_version":    suite.kubeVersion,
+	}
+
+	fullDir := fmt.Sprintf("%s/eks-cluster/", suite.tfDataDir)
+	errTfDir := os.MkdirAll(fullDir, os.ModePerm)
+	suite.Require().NoError(errTfDir)
+
+	tfDir := test_structure.CopyTerraformFolderToDest(suite.T(), "../../modules/", "eks-cluster/", fullDir)
+
+	terraformOptions := &terraform.Options{
+		TerraformDir: tfDir,
+		Upgrade:      false,
+		VarFiles:     []string{"../fixtures/fixtures.default.eks.tfvars"},
+		Vars:         suite.varTf,
+	}
+
+	suite.sugaredLogger.Infow("Creating EKS cluster...", "extraVars", suite.varTf)
+
+	cleanClusterAtTheEnd := utils.GetEnv("CLEAN_CLUSTER_AT_THE_END", "true")
+
+	if cleanClusterAtTheEnd == "true" {
+		defer terraform.Destroy(suite.T(), terraformOptions)
+		defer runtime.HandleCrash(func(i interface{}) {
+			terraform.Destroy(suite.T(), terraformOptions)
+		})
+	}
+
+	terraform.InitAndApplyAndIdempotent(suite.T(), terraformOptions)
+
+	// Wait for the worker nodes to join the cluster
+	sess, err := utils.GetAwsClient()
+	suite.Require().NoErrorf(err, "Failed to get aws client")
+
+	// list your services here
+	eksSvc := eks.NewFromConfig(sess)
+
+	inputEKS := &eks.DescribeClusterInput{
+		Name: aws.String(suite.clusterName),
+	}
+
+	result, err := eksSvc.DescribeCluster(context.Background(), inputEKS)
+	suite.Assert().NoError(err)
+
+	suite.sugaredLogger.Infow("Waiting for worker nodes to join the EKS cluster")
+	errClusterReady := utils.WaitUntilKubeClusterIsReady(result.Cluster, 5*time.Minute, uint64(suite.expectedNodes))
+	suite.Require().NoError(errClusterReady)
+
+	suite.Assert().Equal(suite.kubeVersion, *result.Cluster.Version)
+
+	utils.GenerateKubeConfigFromAWS(suite.T(), suite.region, suite.clusterName, utils.GetAwsProfile(), suite.kubeConfigPath)
+
+	// test suite: deploy a pod and check it is healthy
+	namespace := "example"
+	kubeCtlOptions := k8s.NewKubectlOptions("", suite.kubeConfigPath, namespace)
+	utils.CreateIfNotExistsNamespace(suite.T(), kubeCtlOptions, namespace)
+
+	// deploy the postgres-client Job to test the connection
+	k8s.KubectlApply(suite.T(), kubeCtlOptions, "../../modules/fixtures/whoami-deployment.yml")
+
+	k8s.WaitUntilServiceAvailable(suite.T(), kubeCtlOptions, "whoami-service", 60, 1*time.Second)
+	// wait to ensure service available
+	time.Sleep(30 * time.Second)
+
+	// Now we verify that the service will successfully boot and start serving requests
+	localPort1 := 8883
+
+	service := k8s.GetService(suite.T(), kubeCtlOptions, "whoami-service")
+	portForwardProc1 := k8s.NewTunnel(kubeCtlOptions, k8s.ResourceTypeService, service.ObjectMeta.Name, localPort1, 80)
```

Comment:
since you don't use the localPort1 anywhere else nor rely on it in any way, you can also consider just setting it in the parameter as `0` since then terratest will automatically determine an available port on the host system.